goal_src/jak2/engine/game/game-info.gc

```diff
@@ -5,13 +5,1729 @@
+             )
+            ((= a2-22 14)
+             "bush-port"
+             )
+            ((= a2-22 12)
+             "bush-race-2"
+             )
+            ((= a2-22 9)
+             "whack"
+             )
+            ((= a2-22 1)
+             "race-1"
+             )
+            ((= a2-22 11)
+             "bush-race-1"
+             )
+            ((= a2-22 4)
+             "gungame-red"
+             )
+            ((= a2-22 6)
+             "gungame-blue"
+             )
+            ((= a2-22 5)
+             "gungame-yellow"
+             )
+            ((= a2-22 15)
+             "bush-errol"
+             )
+            ((= a2-22 2)
+             "race-2"
+             )
+            ((zero? a2-22)
+             "none"
+             )
+            ((= a2-22 10)
+             "judge-skatea"
+             )
+            ((= a2-22 17)
+             "reverse-race-2"
+             )
+            ((= a2-22 18)
+             "reverse-race-3"
+             )
+            ((= a2-22 13)
+             "bush-race-3"
+             )
+            ((= a2-22 8)
+             "onin-game"
+             )
+            ((= a2-22 3)
+             "race-3"
+             )
+            (else
+              "*unknown*"
+              )
+            )
+          (-> v1-71 0)
+          (-> v1-71 1)
+          (-> v1-71 2)
+          )
+        )
+      )
+    )
+  (when (= arg0 'entity-perm)
+    (format #t "~Tentity-perm:~%")
+    (let ((game-perms (-> obj perm-list)))
+      (dotimes (s4-4 (-> game-perms length))
+        (format #t "~T~T~`entity-perm`P~%" (-> game-perms data s4-4))
+        )
+      )
+    )
+  obj
+  )
+
+(defmethod you-suck-stage game-info ((obj game-info) (arg0 symbol))
+  (cond
+    ((logtest? (-> *game-info* secrets) (game-secrets hero-mode))
+     0
+     )
+    (else
+      (let ((game-subtasks (-> *game-info* sub-task-list)))
+        (dotimes (i (-> game-subtasks length))
+          (when (nonzero? i)
+            (let ((subtask (-> game-subtasks i)))
+              (when (open? subtask)
+                (let ((cur-lev (level-get-target-inside *level*)))
+                  (when (and cur-lev (= (-> cur-lev info taskname) (-> subtask level)))
+                    (let* ((suck-death-count (if (zero? (-> subtask suck-death-count))
+                                                 5
+                                                 (the-as int (-> subtask suck-death-count))
+                                                 )
+                                             )
+                           (suck-death-stage (/ (the float (-> subtask death-count)) (the float suck-death-count)))
+                           )
+                      (return (min 4 (the int suck-death-stage)))
+                      )
+                    )
+                  )
+                )
+              )
+            )
+          )
+        )
+      0
+      )
+    )
+  )
+
+(defmethod you-suck-scale game-info ((obj game-info))
```

Comment:
interesting, this looks the same as `get-health-percent-lost` in jak 1 (I have no idea where I got that name from)